tests/ui/parser/issues/issue-67377-invalid-syntax-in-enum-discriminant.stderr

```diff
@@ -1,47 +1,27 @@
-error: mismatched closing delimiter: `]`
-  --> $DIR/issue-67377-invalid-syntax-in-enum-discriminant.rs:5:27
-   |
-LL |         V = [PhantomData; { [ () ].len() ].len() as isize,
-   |             -             ^              ^ mismatched closing delimiter
-   |             |             |
-   |             |             unclosed delimiter
-   |             closing delimiter possibly meant for this
-
-error: mismatched closing delimiter: `]`
-  --> $DIR/issue-67377-invalid-syntax-in-enum-discriminant.rs:12:24
-   |
-LL |         V = [Vec::new; { [].len()  ].len() as isize,
-   |             -          ^           ^ mismatched closing delimiter
-   |             |          |
-   |             |          unclosed delimiter
-   |             closing delimiter possibly meant for this
-
-error: mismatched closing delimiter: `]`
-  --> $DIR/issue-67377-invalid-syntax-in-enum-discriminant.rs:19:24
-   |
-LL |         V = [Vec::new; { [0].len() ].len() as isize,
```

Comment:
this is a regression, the old diagnostic reported the positions we need to care about, while the new diagnostic only reported `missing open `[` for this delimiter`, user need to look back which span need to be fixed.